Fix for issue-4612

vbusch opened the pull request:
### Type of change
- Bugfix

### Description
When the router creates a queue on the broker it should be durable

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

Signed-off-by: Vanessa Busch <vbusch@redhat.com>

k-wall commented:
@vbusch rather than the approach you've taken (which I'm not sure would address the whole issue), I think we need to organise that agent always creates the broker resources.   @lulf gave me some ideas on depriving the router messaging user of the create permissions in order to prevent this defect happing (which is the way he's solved the problem in shared).  I'll experiment with this approach this morning.

vbusch commented:
@k-wall great.  I had tested it with the wrong image.  And was thrilled when everything worked.  Even after I started removing extra lines that I thought I didn't need.  After creating the PR, I realized my mistake, and removed you as reviewer and switched it to draft, because I was back where I started.

vbusch commented:
This won't work.  Replaced by #4739
